mlflow/webhooks/dispatch.py

```diff
@@ -59,3 +91,53 @@ def dispatch_webhook(
             f"Failed to dispatch webhook for event {event}: {e}",
             exc_info=True,
         )
+
+
+def test_webhook(webhook: Webhook, event: Optional[WebhookEvent] = None) -> WebhookTestResult:
+    """Test a webhook by sending a test payload.
+
+    Args:
+        webhook: The webhook object to test
+        event: Optional event type to test. If not specified, uses the first event from webhook.
+
+    Returns:
+        WebhookTestResult indicating success/failure and response details
+    """
+    try:
+        # Use provided event or the first event type for testing
+        test_event = event or webhook.events[0]
+
+        # Generate example payload based on the event type
+        if test_event == WebhookEvent.REGISTERED_MODEL_CREATED:
+            from mlflow.webhooks.types import RegisteredModelCreatedPayload
+
+            test_payload = RegisteredModelCreatedPayload.example()
+        elif test_event == WebhookEvent.MODEL_VERSION_CREATED:
+            from mlflow.webhooks.types import ModelVersionCreatedPayload
+
+            test_payload = ModelVersionCreatedPayload.example()
+        elif test_event == WebhookEvent.MODEL_VERSION_TAG_SET:
+            from mlflow.webhooks.types import ModelVersionTagSetPayload
+
+            test_payload = ModelVersionTagSetPayload.example()
+        elif test_event == WebhookEvent.MODEL_VERSION_TAG_DELETED:
+            from mlflow.webhooks.types import ModelVersionTagDeletedPayload
+
+            test_payload = ModelVersionTagDeletedPayload.example()
+        elif test_event == WebhookEvent.MODEL_VERSION_ALIAS_CREATED:
+            from mlflow.webhooks.types import ModelVersionAliasCreatedPayload
+
+            test_payload = ModelVersionAliasCreatedPayload.example()
+        elif test_event == WebhookEvent.MODEL_VERSION_ALIAS_DELETED:
+            from mlflow.webhooks.types import ModelVersionAliasDeletedPayload
+
+            test_payload = ModelVersionAliasDeletedPayload.example()
+        else:
+            raise ValueError(f"Unknown event type: {test_event}")
+
+        return _send_webhook_request(webhook.url, test_payload, webhook.secret)
+    except Exception as e:
+        return WebhookTestResult(
+            success=False,
+            error_message=f"Failed to test webhook: {str(e)[:500]}",
```

Comment:
q: do we need to truncate the error message again?

Reply:
I removed truncation. I'm not sure if we need it. If we include the full traceback, we should consider truncation since it can be long.